File: LeetCode_Python/217_containsDuplicate.py
from typing import List

class ContainsDuplicate:
    def containsDuplicate(self, nums : List[int]) -> bool:
        numsSet = set()

        for num in nums:
            if num in numsSet:
                return True
            else:
                numsSet.add(num)
        return False
    
if __name__ == "__main__":
    solution = ContainsDuplicate()
    nums = [1, 2, 3, 4, 5, 6, 5]

    print(solution.containsDuplicate(nums))

File: LeetCode_Java/217_containsDuplicate.java
import java.util.*;

class ContainsDuplicate{
    public boolean containsDuplicate(int[] nums){
        HashSet<Integer> numsSet = new HashSet<>();

        for(int num : nums){
            if(numsSet.contains(num)){
                return true;
            }
            numsSet.add(num);
        }

        return false;
    }

    public static void main(String[] args){
        ContainsDuplicate solution = new ContainsDuplicate();

        int[] nums = {1, 2, 3, 4, 5, 6, 5};

        System.out.println(solution.containsDuplicate(nums));

    }
}